File: src/ShopNet/Admin/User/Post.ts
import { AdminNet } from "../AdminNet";

/**
 * @author Aloento
 * @since 0.5.0
 * @version 0.1.0
 */
export class AdminUserPost extends AdminNet {
  /**
   * @author Aloento
   * @since 0.5.0
   * @version 0.1.0
   */
  public static async Admin(userId: string): Promise<true> {
    const res = await this.Invoke<true>("UserPostAdmin", userId);
    return res;
  }
}

File: TSystems.LoveOTC/AdminHub/User/Post.cs
namespace TSystems.LoveOTC.AdminHub;

using Helpers;
using Microsoft.EntityFrameworkCore;

internal partial class AdminHub {
    /**
     * <remarks>
     * @author Aloento
     * @since 0.1.0
     * @version 0.2.0
     * </remarks>
     */
    public async Task<bool> UserPostAdmin(Guid userId) {
        var row = await this.Db.Users
            .Where(x => x.UserId == userId)
            .ExecuteUpdateAsync(x =>
                x.SetProperty(u => u.Admin, true));

        this.Logger.GrantAdmin(this.Name, userId, this.Context);
        return row == 1;
    }
}
